server/neptune/workflows/internal/activities/deployment/store.go

```diff
@@ -0,0 +1,132 @@
+package deployment
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/graymeta/stow"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deployment"
+	"github.com/uber-go/tally/v4"
+)
+
+const OutputPrefix = "deployments"
+
+func NewStore(deploymentCfg valid.Deployments, scope tally.Scope) (*instrumentedStore, error) {
+	if deploymentCfg.StorageBackend == nil {
+		return nil, errors.New("error initializing deployment info store")
+	}
+
+	config := deploymentCfg.StorageBackend.BackendConfig.GetConfigMap()
+	backend := deploymentCfg.StorageBackend.BackendConfig.GetConfiguredBackend()
+	containerName := deploymentCfg.StorageBackend.BackendConfig.GetContainerName()
+
+	location, err := stow.Dial(backend, config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &instrumentedStore{
+		store: &store{
+			location:      location,
+			containerName: containerName,
+		},
+		scope: scope.SubScope("store"),
```

Comment:
What is the full metric in this case just out curiosity here?

Reply:
It'd be `atlantis.deploy.store.read_failure` for reading from s3.